webapp/src/cmds.ts

```diff
@@ -447,12 +447,19 @@ export function maybeReconnectAsync(pairIfDeviceNotFound = false, skipIfConnecte
     return reconnectPromise;
 }
 
-export function pairAsync(): Promise<void> {
+export function pairAsync(implicitlyCalled?: boolean): Promise<void> {
     pxt.tickEvent("cmds.pair")
-    return pxt.commands.webUsbPairDialogAsync(pxt.usb.pairAsync, core.confirmAsync)
+    return pxt.commands.webUsbPairDialogAsync(pxt.usb.pairAsync, core.confirmAsync, implicitlyCalled)
         .then(res => {
-            if (res) return maybeReconnectAsync();
-            else return core.infoNotification("Oops, no device was paired.")
+            switch (res) {
+                case pxt.commands.WebUSBPairResult.Success:
+                    return maybeReconnectAsync();
+                case pxt.commands.WebUSBPairResult.Failed:
+                    return core.infoNotification("Oops, no device was paired.")
+                case pxt.commands.WebUSBPairResult.UserRejected:
+                    // User exited pair flow intentionally
```

Comment:
Does this happen for both hitting the close button and for clicking outside the dialog?

Reply:
This one triggers when we decide not to show the pairing automatically in the future -- so right now this will only be if the user clicks the 'download as file' button on the last screen, if they press close or click outside it will show the 'no device paired' warning in the case above, but we have some updates for another pr to clean up the flow / prettify it and we'll need to play around with how much we want to keep showing this dialog in the future